src/lib/storage/table.cpp

```diff
@@ -17,53 +19,103 @@
 
 namespace opossum {
 
+Table::Table(const uint32_t chunk_size) {
+  _max_chunk_size = chunk_size;
+  _chunks.push_back(std::make_shared<Chunk>());
+}
+
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(this->is_empty(), "Columns may only be added to an empty table.");
+  auto search = _column_name_mapping.find(name);
+  DebugAssert(search == _column_name_mapping.end(), "Column names may not be duplicated");
+  _column_name_mapping.insert({name, (ColumnID)_column_names.size()});
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type);
+  _chunks.front()->add_segment(segment);
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
-}
+  // Check if the latest Chunk has already reached maximum size and create a new one if necessary
+  if (this->is_new_chunk_needed()) {
+    _chunks.push_back(std::make_shared<Chunk>());
 
-uint16_t Table::column_count() const {
-  // Implementation goes here
-  return 0;
+    for (auto type_name : _column_types) {
+      auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type_name);
+      _chunks.back()->add_segment(segment);
+    }
+  }
+
+  _chunks.back()->append(values);
 }
 
+bool Table::is_new_chunk_needed() { return _chunks.back()->size() >= _max_chunk_size; }
+
+uint16_t Table::column_count() const { return _column_names.size(); }
+
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  // Since we cannot delete data, all old chunks will have maximum chunk size
```

Comment:
this might be valid for chunks that store the actual data in the first place. but due to selections, chunks with less entries could be produced during query execution. better calculate the size by summing up all individual chunk sizes

Reply:
Will Fix it, when needed for the next sprint. Thanks for the comment though